### PR TITLE
[DOC-9193] Add more details about implicit auth on Dedicated

### PR DESCRIPTION
**Pull request by mdlinville**
[DOC-9193] Add more details about implicit auth on Dedicated

### Preview
[src/current/v23.2/change-data-capture-overview.md](https://deploy-preview-18055--cockroachdb-docs.netlify.app/docs/v23.2/change-data-capture-overview.html)
[src/current/v23.2/cloud-storage-authentication.md](https://deploy-preview-18055--cockroachdb-docs.netlify.app/docs/v23.2/cloud-storage-authentication.html)
[src/current/v23.2/use-cloud-storage.md](https://deploy-preview-18055--cockroachdb-docs.netlify.app/docs/v23.2/use-cloud-storage.html)
[src/current/v23.1/change-data-capture-overview.md](https://deploy-preview-18055--cockroachdb-docs.netlify.app/docs/v23.1/change-data-capture-overview.html)
[src/current/v23.1/cloud-storage-authentication.md](https://deploy-preview-18055--cockroachdb-docs.netlify.app/docs/v23.1/cloud-storage-authentication.html)
[src/current/v23.1/use-cloud-storage.md](https://deploy-preview-18055--cockroachdb-docs.netlify.app/docs/v23.1/use-cloud-storage.html)

### CODE REVIEW

File: src/current/v23.2/cloud-storage-authentication.md [hunk 1]
```diff
@@ -364,6 +364,10 @@ For CockroachDB instances that are running within a Google Cloud Environment, [e
 
 For CockroachDB clusters running in other environments, `implicit` authentication access can still be set up manually with the following steps:
 
+### Use implicit authentication with your own service account
```
Comment (mdlinville) [inline src/current/v23.2/cloud-storage-authentication.md:367]: Here begin the non-whitespace changes.

File: src/current/v23.2/cloud-storage-authentication.md [hunk 2]
```diff
@@ -396,7 +414,7 @@ CockroachDB supports assume role authentication on clusters running v22.2. Authe
...
 
-The access is also limited by the generated [short-lived credentials](https://cloud.google.com/iam/docs/create-short-lived-credentials-direct). The service account/role that is assuming another role will issue the request for the short-lived credentials. If there are multiple roles in the [chain](#google-cloud-storage-role-chaining), then each role defined in the chain will issue the request for credentials for the next role in the chain. 
+The access is also limited by the generated [short-lived credentials](https://cloud.google.com/iam/docs/create-short-lived-credentials-direct). The service account/role that is assuming another role will issue the request for the short-lived credentials. If there are multiple roles in the [chain](#google-cloud-storage-role-chaining), then each role defined in the chain will issue the request for credentials for the next role in the chain.
```
Comment (mdlinville) [inline src/current/v23.2/cloud-storage-authentication.md:417]: Here and following are more whitespace-only changes.

File: src/current/v23.2/cloud-storage-authentication.md [hunk 3]
```diff
@@ -388,6 +392,20 @@ For CockroachDB clusters running in other environments, `implicit` authenticatio
...
+For a CockroachDB {{ site.data.products.dedicated }} cluster, you can either [use implicit authentication with your own service account](#use-implicit-authentication-with-your-own-service-account) or you can follow these instructions to use implicit authentication with the service account that CockroachDB {{ site.data.products.cloud }} automatically creates and manages for each CockroachDB {{ site.data.products.dedicated }} cluster. This service account is named according to the pattern `crl-dr-store-user-{CLUSTER_ID}`, where `{CLUSTER_ID}` is the ID of the cluster, and is automatically granted the permission to write to the cluster's storage. One benefit of this approach is that no additional credentials needs to be added to the cluster.
+
+For example, to initiate a manual backup on a CockroachDB {{ site.data.products.dedicated }} cluster using implicit auth with the `crl-dr-store-user-{CLUSTER_ID}` service account:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+BACKUP INTO 'gs://{bucket name}/{path}?&AUTH=implicit&IAM_ROLE=crl-dr-store-user-{CLUSTER_ID}'
+~~~
+
+Replace `{CLUSTER_ID}` with the cluster's ID.
```
Comment (adwittumuluri) [inline src/current/v23.2/cloud-storage-authentication.md:407]: I simplified a bit in our slack thread. The `{CLUSTER_ID}` is the "the last 12 digits of your cluster's id", as described [here](https://deploy-preview-18055--cockroachdb-docs.netlify.app/docs/dev/cloud-storage-authentication?filters=gcs#set-up-the-identity-service-account-for-dedicated-clusters). It's not the full ID; maybe we should change to `{cluster ID suffix}` for consistency?

File: src/current/v23.2/cloud-storage-authentication.md [hunk 4]
```diff
@@ -388,6 +392,20 @@ For CockroachDB clusters running in other environments, `implicit` authenticatio
...
+### Use implicit authentication with CockroachDB {{ site.data.products.dedicated }}
+
+For a CockroachDB {{ site.data.products.dedicated }} cluster, you can either [use implicit authentication with your own service account](#use-implicit-authentication-with-your-own-service-account) or you can follow these instructions to use implicit authentication with the service account that CockroachDB {{ site.data.products.cloud }} automatically creates and manages for each CockroachDB {{ site.data.products.dedicated }} cluster. This service account is named according to the pattern `crl-dr-store-user-{CLUSTER_ID}`, where `{CLUSTER_ID}` is the ID of the cluster, and is automatically granted the permission to write to the cluster's storage. One benefit of this approach is that no additional credentials needs to be added to the cluster.
+
+For example, to initiate a manual backup on a CockroachDB {{ site.data.products.dedicated }} cluster using implicit auth with the `crl-dr-store-user-{CLUSTER_ID}` service account:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+BACKUP INTO 'gs://{bucket name}/{path}?&AUTH=implicit&IAM_ROLE=crl-dr-store-user-{CLUSTER_ID}'
```
Comment (adwittumuluri) [inline src/current/v23.2/cloud-storage-authentication.md:404]: This flow is also possible with AWS - customers can give `arn:aws:iam::{AWS account ID}:role/crl-dr-store-user-{cluster ID suffix}` direct permissions on an S3 bucket. Is that something to add as part of this PR, or a different one?

**Review by adwittumuluri — CHANGES_REQUESTED** [review]

File: src/current/v23.2/cloud-storage-authentication.md [hunk 5]
```diff
@@ -74,12 +78,37 @@ aws ec2 associate-iam-instance-profile --iam-instance-profile Name={example prof
...
+
+For a CockroachDB {{ site.data.products.dedicated }} cluster, you can either [use implicit authentication with your own IAM role](#use-implicit-authentication-with-your-own-iam-role) or you can follow these instructions to use implicit authentication with the IAM role that CockroachDB {{ site.data.products.cloud }} automatically creates and manages for each CockroachDB {{ site.data.products.dedicated }} cluster. This IAM role is named according to the pattern `arn:aws:iam::{AWS_ACCOUNT_ID}:role/crl-dr-store-user-{CLUSTER_ID_SUFFIX}`, where `{CLUSTER_ID_SUFFIX}` is the last 12 digits of the cluster's ID. This IAM role is automatically granted the permission to write to the cluster's storage. One benefit of this approach is that no additional credentials needs to be added to the cluster.
+
+For example, to initiate a manual backup on a CockroachDB {{ site.data.products.dedicated }} cluster using implicit auth with the `crl-dr-store-user-{CLUSTER_ID_SUFFIX}` IAM role:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+BACKUP INTO 's3://{bucket name}/{path}?&AUTH=implicit&IAM_ROLE=crl-dr-store-user-{CLUSTER_ID_SUFFIX}'
```
Comment (adwittumuluri) [inline src/current/v23.2/cloud-storage-authentication.md:90]: Do we want to expand the `IAM_ROLE` more in this example query? It'd need to be the full `arn:aws:iam::{AWS_ACCOUNT_ID}:role/crl-dr-store-user-{CLUSTER_ID_SUFFIX}` ARN that you mentioned above, not just `crl-dr-store-{CLUSTER_ID_SUFFIX}`

File: src/current/v23.2/cloud-storage-authentication.md [hunk 6]
```diff
@@ -74,12 +78,37 @@ aws ec2 associate-iam-instance-profile --iam-instance-profile Name={example prof
...
 {{site.data.alerts.end}}
 
+For a CockroachDB {{ site.data.products.dedicated }} cluster, you can either [use implicit authentication with your own service account](#use-implicit-authentication-with-your-own-service-account) or you can follow these instructions to use implicit authentication with the service account that CockroachDB {{ site.data.products.cloud }} automatically creates and manages for each CockroachDB {{ site.data.products.dedicated }} cluster. This service account is named according to the pattern `crl-dr-store-user-{CLUSTER_ID_SUFFIX}`, where `{CLUSTER_ID}` is the last 12 digits of the cluster's. This service account is automatically granted the permission to write to the cluster's storage. One benefit of this approach is that no additional credentials needs to be added to the cluster.
+
+For example, to initiate a manual backup on a CockroachDB {{ site.data.products.dedicated }} cluster using implicit auth with the `crl-dr-store-user-{CLUSTER_ID}` service account:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+BACKUP INTO 'gs://{bucket name}/{path}?&AUTH=implicit&IAM_ROLE=crl-dr-store-user-{CLUSTER_ID_SUFFIX}'
```
Comment (adwittumuluri) [inline src/current/v23.2/cloud-storage-authentication.md:105]: I'm slightly confused by the organization here. This example query and the text before it seem to be related to GCP, but they're under the AWS section. Does this block just have to move to GCP? It seems to be rendered under AWS:
image

File: src/current/v23.2/cloud-storage-authentication.md [hunk 7]
```diff
@@ -388,15 +421,29 @@ For CockroachDB clusters running in other environments, `implicit` authenticatio
...
+
+For a CockroachDB {{ site.data.products.dedicated }} cluster, you can either [use implicit authentication with your own service account](#use-implicit-authentication-with-your-own-service-account) or you can follow these instructions to use implicit authentication with the service account that CockroachDB {{ site.data.products.cloud }} automatically creates and manages for each CockroachDB {{ site.data.products.dedicated }} cluster. This service account is named according to the pattern `crl-dr-store-user-{CLUSTER_ID_SUFFIX}`, where `{CLUSTER_ID}` is the last 12 digits of the cluster's. This service account is automatically granted the permission to write to the cluster's storage. One benefit of this approach is that no additional credentials needs to be added to the cluster.
+
+For example, to initiate a manual backup on a CockroachDB {{ site.data.products.dedicated }} cluster using implicit auth with the `crl-dr-store-user-{CLUSTER_ID}` service account:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+BACKUP INTO 'gs://{bucket name}/{path}?&AUTH=implicit&IAM_ROLE=crl-dr-store-user-{CLUSTER_ID_SUFFIX}'
```
Comment (adwittumuluri) [inline src/current/v23.2/cloud-storage-authentication.md:433]: similar to the example with AWS, the full service account is `crl-dr-store-user-{CLUSTER_ID_SUFFIX}@{PROJECT_ID}.iam.gserviceaccount.com`, similar to [here](https://www.cockroachlabs.com/docs/cockroachcloud/cmek-ops-gcp#step-2-provision-the-cross-tenant-service-account).

File: src/current/v23.2/cloud-storage-authentication.md (hunk 6 above)
Comment (adwittumuluri) [reply src/current/v23.2/cloud-storage-authentication.md:105]: Or actually, it seems like this block similar to the GCP additions below, so maybe its a duplicate?

**Review by adwittumuluri — CHANGES_REQUESTED** [review]

File: src/current/v23.2/cloud-storage-authentication.md (hunk 6 above)
Comment (mdlinville) [reply src/current/v23.2/cloud-storage-authentication.md:105]: I may have done something wrong that is making things show up incorrectly. This page uses "filter tabs" and the AWS content is supposed to show up in the AWS tab, the GCP content in the GCP tab. Each tab is within a `<section>` with specific syntax. Let me make sure I didn't do something wrong.

Comment (mdlinville) [reply src/current/v23.2/cloud-storage-authentication.md]: I verified that this is working as expected. If you select AWS, you should see the AWS content including the new section. If you select GCP, you see the new GCP content. PTAL.

File: src/current/v23.2/cloud-storage-authentication.md [hunk 8]
```diff
@@ -388,17 +405,22 @@ For CockroachDB clusters running in other environments, `implicit` authenticatio
...
+### Use implicit authentication with CockroachDB {{ site.data.products.dedicated }}
 
-{{site.data.alerts.callout_info}}
-CockroachDB supports assume role authentication on clusters running v22.2. Authenticating to cloud storage with `ASSUME_ROLE` on clusters running versions v22.1 and earlier, or mixed versions, is not supported and will result in failed bulk operations.
-{{site.data.alerts.end}}
+For a CockroachDB {{ site.data.products.dedicated }} cluster, you can either [use implicit authentication with your own service account](#use-implicit-authentication-with-your-own-service-account) or you can follow these instructions to use implicit authentication with the service account that CockroachDB {{ site.data.products.cloud }} automatically creates and manages for each CockroachDB {{ site.data.products.dedicated }} cluster. This service account is named according to the pattern `crl-dr-store-user-{CLUSTER_ID_SUFFIX}`, where `{CLUSTER_ID}` is the last 12 digits of the cluster's. This service account is automatically granted the permission to write to the cluster's storage. One benefit of this approach is that no additional credentials needs to be added to the cluster.
```
Comment (adwittumuluri) [inline src/current/v23.2/cloud-storage-authentication.md:411]: nit: not sure if you wanted to include `@{PROJECT_ID}.iam.gserviceaccount.com` in this text considering you have the full ARN for the AWS example.

**Review by adwittumuluri — APPROVED** [review]

**Review by kathancox — APPROVED** [review]
Just a couple of small nits, but otherwise lgtm

File: src/current/v23.1/cloud-storage-authentication.md [hunk 9]
```diff
@@ -74,21 +84,17 @@ aws ec2 associate-iam-instance-profile --iam-instance-profile Name={example prof
...
 
-{{site.data.alerts.callout_success}}
-Role assumption applies the principle of least privilege rather than directly providing privilege to a user. Creating IAM roles to manage access to AWS resources is Amazon's recommended approach compared to giving access directly to IAM users.
-{{site.data.alerts.end}}
+For example, to initiate a manual backup on a CockroachDB {{ site.data.products.dedicated }} cluster using implicit auth with the `crl-dr-store-user-{CLUSTER_ID_SUFFIX}` IAM role:
 
-The [following section](#set-up-amazon-s3-assume-role) demonstrates setting up assume role authentication between two users. To see examples of assume role authentication for more than two roles, see [Role chaining](#amazon-s3-role-chaining) for additional detail.
+{% include_cached copy-clipboard.html %}
+~~~ sql
+BACKUP INTO 's3://{bucket name}/{path}?&AUTH=implicit&IAM_ROLE=arn:aws:iam::{AWS_ACCOUNT_ID}:role/crl-dr-store-user-{CLUSTER_ID_SUFFIX}`
```
Comment (mdlinville) [inline src/current/v23.1/cloud-storage-authentication.md:96]:
```suggestion
BACKUP INTO 's3://{bucket name}/{path}?&AUTH=implicit&IAM_ROLE=arn:aws:iam::{AWS_ACCOUNT_ID}:role/crl-dr-store-user-{CLUSTER_ID_SUFFIX}`;
```

File: src/current/v23.1/cloud-storage-authentication.md [hunk 10]
```diff
@@ -412,6 +427,23 @@ For CockroachDB clusters running in other environments, `implicit` authenticatio
...
+
+For example, to initiate a manual backup on a CockroachDB {{ site.data.products.dedicated }} cluster using implicit auth with the `crl-dr-store-user-{CLUSTER_ID}` service account:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+BACKUP INTO 'gs://{bucket name}/{path}?&AUTH=implicit&IAM_ROLE=crl-dr-store-user-{CLUSTER_ID_SUFFIX}@{PROJECT_ID}.iam.gserviceaccount.com'
```
Comment (mdlinville) [inline src/current/v23.1/cloud-storage-authentication.md:439]:
```suggestion
BACKUP INTO 'gs://{bucket name}/{path}?&AUTH=implicit&IAM_ROLE=crl-dr-store-user-{CLUSTER_ID_SUFFIX}@{PROJECT_ID}.iam.gserviceaccount.com';
```

File: src/current/v23.1/cloud-storage-authentication.md [hunk 11]
```diff
@@ -47,13 +47,23 @@ BACKUP DATABASE <database> INTO 's3://{bucket name}/{path in bucket}/?AWS_ACCESS
...
 ## Amazon S3 implicit
 
 {{site.data.alerts.callout_info}}
-To use `implicit` authentication on a CockroachDB {{ site.data.products.cloud }} cluster, it is necessary to authenticate using assume role authentication. See [Amazon S3 assume role](#amazon-s3-assume-role) for more details.
+To use `implicit` authentication on a CockroachDB {{ site.data.products.cloud }} cluster, it is necessary to authenticate using assume role authentication. See [Amazon S3 assume role](#set-up-amazon-s3-assume-role) for more details.
+{{site.data.alerts.end}}
+
+### Use implicit authentication with your own IAM role
+
+For self-hosted deployments, follow these instructions. For CockroachDB Dedicated deployments, you can either follow these instructions or you can [use implicit authentication together with the service account which CockroachDB Cloud automatically creates](/docs/v23.2/cloud-storage-authentication.html?filters=aws#use-implicit-authentication-dedicated).
```
Comment (mdlinville) [inline src/current/v23.1/cloud-storage-authentication.md:61]:
```suggestion
For self-hosted deployments, follow these instructions. For CockroachDB Dedicated deployments, you can either follow these instructions or you can [use implicit authentication together with the service account that CockroachDB Cloud automatically creates](/docs/v23.2/cloud-storage-authentication.html?filters=aws#use-implicit-authentication-dedicated).
```

File: src/current/v23.1/cloud-storage-authentication.md [hunk 12]
```diff
@@ -378,16 +384,25 @@ BACKUP DATABASE <database> INTO 'gs://{bucket name}/{path}?AUTH=specified&CREDEN
...
 
+To use `implicit` authentication on a CockroachDB {{ site.data.products.cloud }} cluster, it is necessary to authenticate using assume role authentication. See [Set up Google Cloud Storage assume role](#set-up-google-cloud-storage-assume-role) for more details about configuring the role and service account.
+
 {% include {{ page.version.version }}/misc/external-io-privilege.md %}
 
 For CockroachDB clusters running in other environments, `implicit` authentication access can still be set up manually with the following steps:
 
+### Use implicit authentication with your own service account
+
+For self-hosted deployments, follow these instructions. For CockroachDB {{ site.data.products.dedicated }} deployments, you can either follow these instructions or you can [use implicit authentication together with the service account which CockroachDB {{ site.data.products.cloud }} automatically creates](#use-implicit-authentication-dedicated).
```
Comment (mdlinville) [inline src/current/v23.1/cloud-storage-authentication.md:404]:
```suggestion
For self-hosted deployments, follow these instructions. For CockroachDB {{ site.data.products.dedicated }} deployments, you can either follow these instructions or you can [use implicit authentication together with the service account that CockroachDB {{ site.data.products.cloud }} automatically creates](#use-implicit-authentication-dedicated).
```

File: src/current/v23.2/cloud-storage-authentication.md [hunk 13]
```diff
@@ -379,15 +414,24 @@ BACKUP DATABASE <database> INTO 'gs://{bucket name}/{path}?AUTH=specified&CREDEN
...
 
+To use `implicit` authentication on a CockroachDB {{ site.data.products.cloud }} cluster, it is necessary to authenticate using assume role authentication. See [Set up Google Cloud Storage assume role](#set-up-google-cloud-storage-assume-role) for more details about configuring the role and service account.
+
 {% include {{ page.version.version }}/misc/external-io-privilege.md %}
 
 For CockroachDB clusters running in other environments, `implicit` authentication access can still be set up manually with the following steps:
 
+### Use implicit authentication with your own service account
+
+For self-hosted deployments, follow these instructions. For CockroachDB {{ site.data.products.dedicated }} deployments, you can either follow these instructions or you can [use implicit authentication together with the service account which CockroachDB {{ site.data.products.cloud }} automatically creates](#use-implicit-authentication-dedicated).
```
Comment (mdlinville) [inline src/current/v23.2/cloud-storage-authentication.md:433]:
```suggestion
For self-hosted deployments, follow these instructions. For CockroachDB {{ site.data.products.dedicated }} deployments, you can either follow these instructions or you can [use implicit authentication together with the service account that CockroachDB {{ site.data.products.cloud }} automatically creates](#use-implicit-authentication-dedicated).
```

**Pull request merged**